deepmd/pt/infer/deep_eval.py

```diff
@@ -109,9 +207,19 @@ def eval(
         natoms, numb_test = self._get_natoms_and_nframes(
             coords, atom_types, len(atom_types.shape) > 1
         )
-        return self._eval_func(self._eval_model, numb_test, natoms)(
+        out = self._eval_func(self._eval_model, numb_test, natoms)(
             coords, cells, atom_types, atomic
         )
+        return dict(
+            zip(
+                [
+                    x.name
+                    for x in self.output_def.var_defs.values()
+                    if atomic or "_redu" in x.name or "_derv_r" in x.name
```

Comment:
_derv_r is atomic?

Reply:
what is the condition actually is ? provide a method to get the condition out of a model output def?

Reply:
When atomic is True, all outputs are required; otherwise, only energy, force, and virial are required.

Reply:
map all outputs and pop the atomic energy and virial when `atomic==False` ? 
or provide a method like `get_reduced_and_force`

Reply:
I added a method called `_get_request_defs` and directly passed `request_defs` (instead of `atomic`) to the inner method.